### ISSUE TITLE
Two way sync - Set all events as private

### ISSUE THREAD

**Issue by phw198** [phw198/OutlookGoogleCalendarSync#2]
Hello,  
  
Is it possible to add the feature that all events are marked as "Private" when
syncing two way?  
  
I'm currently sync'ing my work calendar with my client calendar and want all
my work events to be private, so that my client can't see my work meeting
details.  

Work Item Details
--------------------
**Original CodePlex Issue:**	[Issue 88](http://outlookgooglecalendarsync.codeplex.com/workitem/88)
**Status:**	Proposed
**Reason Closed:**	Unassigned
**Assigned to:**	Unassigned
**Reported on:**	Sep 9, 2015 at 1:11 PM
**Reported by:**	Silenceofdk
**Updated on:**	Jun 16, 2016 at 6:38 AM
**Updated by:**	Silenceofdk

**Comment by phw198** [comment]
On *2015-09-21 05:19:15 UTC*, **phw198** commented:

Thanks for the suggestion - I'm sure it could be possible. 

Just so I'm clear though, would it work to create a second "silenceofdk at work" calendar, sync to that as normal and then share that calendar with your client - but only let them see your free/busy status?

**Comment by phw198** [comment]
On *2015-09-26 14:36:35 UTC*, **Silenceofdk** commented:

My calenders with clients always look like this:

My Work(ExchangeWork - EW) | Google Mail | Client(ExchangeClient - EC)

Visibility should be that all items synced from Google mail to EC should have visibility "hidden". It isn't need for items to Google mail or EW.

I did see some flag items regarding privacy, perhaps one can make a option with something there?

**Comment by phw198** [comment]
On *2015-09-26 17:38:10 UTC*, **phw198** commented:

So who is doing the syncing from Google Mail to EC? Is that you with OGCS? Ie.

EW -> (you via OGCS) -> GMail -> (you via OGCS) -> EC

If so, is the EC Exchange Mailbox your own account, or a shared mailbox with other clients? If it's your own, can should be able to set the permissions on that Outlook Calendar so others can only see your free/busy status.

Syncing items as private to Outlook would be useless if people at your client office have direct access to your Exchange mailbox anyway - it's only private with the details hidden to those with whom you have shared the calendar.

Let me know if I'm still not quite getting what you need to do...

**Comment by phw198** [comment]
On *2015-10-26 17:06:12 UTC*, **Silenceofdk** commented:

Sorry about the delay in writing back. Yes the syncing from Gmail to EC is me with OGCS and it's my own account, e.g. not shared. So yes you are correct(However it's two-way) in that: EW <-> (you via OGCS) <-> GMail <-> (you via OGCS)
 <-> EC.

You are correct, I could change the permissions in Outlook Calendar, however that would prevent the client from seeing all meetings even at the client location, which they are allowed to see(And I have had one client bitch about everything being private).


Basically, I just want to hide my EW meetings and details from the client when they look in my EC, so that all EW meetings have the "Private" ticked, as these something contain confidential information or meetings about the client and possible leads.
 You don't want a client stumbling upon that :)

**Comment by phw198** [comment]
On *2015-10-26 19:40:59 UTC*, **phw198** commented:

Hmm, OK I see your point. Problem is you're basically asking for two-way sync, but to force Private only in 1 direction. Do-able, but will need to think on it a bit...

BTW, have you seen the obfuscation option? It only works on the subject (so don't sync the description), but you can mask words/phrases with regular expressions. This might be enough for what you need? Worse case you a regex that changes the subject to "Private"
 and don't sync attendees etc.

**Comment by phw198** [comment]
On *2016-06-16 13:38:45 UTC*, **Silenceofdk** commented:

Hi pwh198

Would it be easier/possible to force the "Private" flag in only one direction? I'm thinking you then just use the 1-way sync both ways.

**Comment by phw198** [comment]
- [x] Update interface and settings
- [x] Implement logic in sync engine
- [x] Release test build

**Comment by phw198** [comment]
:package: Pre-alpha build for those that want to try this feature out with [v2.5.1.1.zip](https://github.com/phw198/OutlookGoogleCalendarSync/files/1140473/v2.5.1.1.zip). 

The option is under Settings > Sync Options > How > More...

See [instructions for applying](https://github.com/phw198/OutlookGoogleCalendarSync/wiki/Reporting-Problems#applying-a-hotfix) in the same way as a hotfix. Please **ensure you are already running v2.5.1.0** before applying this.

Let me know how it goes. :speech_balloon:

**Comment by fors6630** [comment]
You rock Phil - thanks!  2.5.1.1 is installed and will let you know what I find.  

One really minor item I think I just observed, in case it helps: 
1) Target calendar item (work) owned / created a meeting that was sent to source calendar item as invitee (home).  Availability was set to be shown as busy.  
2) Target calendar item then was changed Availability = free.  
2) Upon triggering the sync in 2.5.1.1, target calendar item was updating to again be busy, which is incorrect and likely due to the circular relationship use case I tried to describe here.  

So minor, but could lead to other issue so wanted you to be aware.  

Is there a way I can tell in outlook, what items originated from the gcalsync app?  (i.e., were sync'd from another calendar)

Thanks for all you're doing!

**Comment by fors6630** [comment]
Hi Phil - I just tested this again, after discovering a item created in google calendar was in outlook, but was not private.  I'm running v2.6.1.0, and can confirm that it's not making sync'd events private in outlook even though my setting is set to "set entries as private in target calendar".  

In reviewing the verbose output and logs, I see one error, as well as a bunch of warnings.  I think I should clear out all sync'd items from outlook, prior to doing any further debugging.  

How can I quickly find all the items that originated in google calendar, then were added to outlook/exchange via OGCS?  

Thanks...

**Comment by fors6630** [comment]
I think I figured out how to clean out the calendar, via the user defined OGCS fields.  I did that, re-sync'd, and still have the gcal items showing up as visible so the new setting doesn't seem to be working in this v2.6.1.0.   Am I doing something wrong?

**Comment by phw198** [comment]
I've just tested a one way sync from Google to Outlook and it set the Outlook item as private. Could you provide your OGcalsync.log file? Ultra fine logging level will show each appointment attribute comparison value...

**Comment by fors6630** [comment]
Here is the log file.  Thanks Paul.

**Comment by phw198** [comment]
Thanks for that @fors6630 

:package: This should now be resolved with [hotfix v2.6.1.9.zip](https://github.com/phw198/OutlookGoogleCalendarSync/files/1290971/v2.6.1.9.zip) [[see instructions for applying](https://github.com/phw198/OutlookGoogleCalendarSync/wiki/Reporting-Problems#applying-a-hotfix)].

Please let me know how it goes.

**Comment by fors6630** [comment]
I just installed the hotfix and initial testing looks like you nailed it - thanks Paul - you rock!  :1st_place_medal: 

Now to get the rest of #186 so I'm not accidentally blocking my work calendar with home things.  :)

LMK if I can help further.
